### PR TITLE
Try to use numpy.nan in C function

### PR DESCRIPTION
**Pull request by densmirn**
test case:
`pytest "tests/test_mathematical.py::test_nancumprod[array1]" -s`

### CODE REVIEW

**Review by shssf — APPROVED** [review]

**Comment by densmirn** [comment]
3 more tests passed in comparison to master branch

File: dpnp/backend/CMakeLists.txt [hunk 1]
```diff
@@ -148,6 +148,9 @@ if(NOT WIN32)
   add_link_options(-Wl,-z,noexecstack,-z,relro,-z,now)
 endif()
 
+# used to handle NaN in the user input data
+add_compile_options(-fno-fast-math)
```
Comment (Alexander-Makaryev) [inline dpnp/backend/CMakeLists.txt:152]: Can we implement it without new compiler flags?

**Comment by Alexander-Makaryev** [comment]
@densmirn Could you please add some comment about closing of this PR?

**Comment by densmirn** [comment]
I closed this PR accidently by removing branch. So reopened it.

**Pull request closed without merging**